corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -910,21 +910,48 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
         };
 
         self.file = ko.observable();
-
+        self.extensionsMap = {
+            "image/*,.pdf": ["jpg", "jpeg", "png", "pdf"],
+            "audio/*": ["3ga","mp3", "wav", "amr", "qcp","ogg"],
+            "video/*": ["3gpp", "3gp", "3gp2", "3g2", "mp4","mpg4", "mpeg4", "m4v", "mpg", "mpeg"],
+        };
```

Comment:
This corresponds to `SUPPORTED_FILE_EXTS` which is a list used for validation in Formplayer. 
https://github.com/dimagi/formplayer/pull/1198/files

Reply:
Can you add a comment saying that?

Reply:
@Robert-Costello FP actually does a `Either-Or` [check](https://github.com/dimagi/formplayer/blob/saveMediaAttachment/src/main/java/org/commcare/formplayer/services/MediaHandler.kt#L91) on extension and content-type of a file. So a file doesn't have one of the listed extensions but correspond to right content type should upload without errors. Just want to make sure we are following the same strategy here as well.

Reply:
ah, ok. I'll have to make another change to make it an either/or check on this end.

Reply:
fixed here https://github.com/dimagi/commcare-hq/pull/32634/commits/5109b4966285086f5dc97fafeb552c2df8de6111
I also made a change in FP here https://github.com/dimagi/formplayer/pull/1198/commits/577a0a16097de3fb1e717b6f26c146c1bfd31614 because the `mimeType` var was only ever being set to `null` because `file.name` was always "file".

Reply:
I think a `mp4` can be audio as well but then our either-or check on ext and contentType will pass for mp4 videos in audio questions. So in retrsopect the either-or check doesn't seem like a good fit (we did it on FP because mobile have the same) here, So it's a trade off between consistency and right behaviour. I will go for right behaviour and change it to strictly matching the content type against question type and extensions ? 

Also if we are changing validation here, do we need to care about matching against file extensions at all if the contentType validates succesfully ? 

@snopoke @orangejenny looking for a second opinion on this.

Reply:
Weird, I thought `accept` was implemented at the browser level. But if it's an OS difference, then yeah, I do think we should validate on the front end.

Without the front end validation, the file gets sent to formplayer, but formplayer rejects it, right?

Reply:
> if someone uploaded an inappropriate file type, the risk is that somewhere else in HQ, when they attempt to view the file (which I don't think this is possible now, but if not we'll likely want this functionality in the future), HQ wouldn't display/preview it properly. Is that accurate?

That makes sense. Although what I was thinking is validating against both content type **AND** extension seems like a better behaviour instead of validating contentType **OR** file extension. I earlier commented that we should do `either OR` because FP does the same and mobile does the same but in retrospect think we should change it to `and`.

Reply:
>Without the front end validation, the file gets sent to formplayer, but formplayer rejects it, right?

Yes, if I understand correctly formplayer rejects the file if it doesn't match accepted file extensions and doesn't match accepted mime type. That is a more broad check than this added frontend check, meaning any valid file will be accepted by formplayer regardless of the question type.